src/Validation/Assumes.cs

```diff
@@ -5,78 +5,124 @@ namespace Validation
 {
     using System;
     using System.Collections.Generic;
+    using System.ComponentModel;
     using System.Diagnostics;
-    using System.Globalization;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+    using System.Runtime;
+    using System.Runtime.CompilerServices;
 
     /// <summary>
     /// Common runtime checks that throw public error exceptions upon failure.
     /// </summary>
     public static partial class Assumes
     {
         /// <summary>
-        /// Throws an exception if the specified value is null.
+        /// Throws an public exception.
```

Comment:
Should we change this to `Throws an <see cref="InternalErrorException" />` ?

Reply:
It's more accurate that way, but the exception is an internal class. Should we keep it as internal or make it a public class?